mmbench/workflow/predict.py

```diff
@@ -33,4 +43,97 @@ def benchmark_pred_exp(dataset, datasetdir, outdir):
     same number of samples and subjects, but possibly different latent
     dimensions.
     """
-    raise NotImplementedError
+    print_title("COMPARE MODELS USING REGRESSIONS "
+                f"OR CLASSIFICATION WITH ML ANALYSIS: {dataset}")
+    benchdir = outdir
+    print_text(f"Benchmark directory: {benchdir}")
+
+    print_subtitle("Loading data...")
+
+    latent_data = np.load(os.path.join(benchdir, f"latent_vecs_{dataset}.npz"))
+    latent_data_tr = np.load(os.path.join(benchdir,
+                                          f"latent_vecs_train_{dataset}.npz"))
+    assert latent_data.keys() == latent_data_tr.keys(), (
+        "latent data must have the same keys")
+    shape = None
+    meta_df = pd.read_csv(
+        os.path.join(benchdir, f"latent_meta_{dataset}.tsv"), sep="\t")
+    meta_df_tr = pd.read_csv(
+        os.path.join(benchdir, f"latent_meta_train_{dataset}.tsv"), sep="\t")
+    clinical_scores = meta_df_tr.columns
+    reg = linear_model.Ridge(alpha=.5)
+    cla = linear_model.RidgeClassifier()
+    predict_records, predict_results = dict(), dict()
```

Comment:
use only one dict, and then from use pandas melt to convert table from wide to long format.